pypdf/_writer.py

```diff
@@ -249,7 +249,6 @@ def _get_clone_from(
         # to prevent overwriting
         self.temp_fileobj = fileobj
         self.fileobj = ""
-        self.with_as_usage = False
```

Comment:
I'm not found of removing with_as_usage attribute. it may be usefull to know that the the object has been created for a context manager.

Reply:
It's never used.  Keeping it around would mislead someone reading the code that it matters in some way.  It's dead code but easy to revive if a need arises.

Reply:
I agree that having a dead property/attribute is not good, but it is a public interface which tends to need a proper deprecation process.

Reply:
I'm happy to put `with_as_usage` back if it will reduce friction.

Reply:
Could we please make `with_as_usage` a property which can use `deprecate_no_replacement("with_as_usage", "6.0")`?

Reply:
Done.